taiga_halo2/src/circuit/resource_logic_examples/or_relation_intent.rs

```diff
@@ -220,34 +249,48 @@ impl ResourceLogicCircuit for OrRelationIntentResourceLogicCircuit {
             || "conditional equal: check value",
             |mut region| {
                 config.conditional_equal_config.assign_region(
-                    &is_input_resource,
+                    &self_resource.is_input,
                     &receiver_value,
-                    &basic_variables.output_resource_variables[0]
-                        .resource_variables
-                        .value,
+                    &desired_resource.resource.value,
                     0,
                     &mut region,
                 )
             },
         )?;
 
+        // check the desired_resource is an output
+        {
+            let zero_constant = assign_free_constant(
```

Comment:
Minor suggestion: shall we have a file with zero and one constants? They come up often